cfpb/consumerfinance.gov
# [WIP][Paying for College] Remove jQuery from Disclosures app

This PR will remove jQuery from the P4C "Dynamic Disclosures" app.

Originally, the intent was to add a bunch of cypress tests to verify the app's functionality, then remove all the jQuery, but after a long time trying to figure out why GHA failed all the cypress tests that passed locally, I finally found that it was the jQuery itself that was causing the GHA cypress failures, so we couldn't really write a PR with passing tests while the jQuery was in there.

Thus, this PR adds the cypress tests while also removing the jQuery. Not ideal, but it is easy enough to first verify the cypress tests pass with the existing code, then check again that they pass with this branch. See "How to test this PR" for a quick command to copy/paste to test the cypress in our `main` branch locally.

---

## Additions
- Add cypress tests for Dynamic Disclosures
- Add `dollar-sign.js` which is a vanilla replacement for some jQuery functions


## Removals
- removed jQuery!

## Changes
- Change some jQuery calls to vanilla JS calls

## How to test this PR
1. Pull down the `disclosures` branch
2. From your `main` branch, you can checkout just the cypress tests:
  `git checkout disclosures -- test/cypress/fixtures/paying-for-college/ test/cypress/integration/paying-for-college/disclosures/`
3. Run the cypress tests (under `paying-for-college/disclosures` in your cypress interface) against the `main` branch, where jQuery still reigns supreme. (Make sure your JS files are built in this branch before testing!)
4. Then go to the `disclosures` branch, rebuild your JS files, and then try the cypress tests again!
5. Check out the general functionality of the disclosures app. Here's a test URL: [test url for localhost](http://localhost:8000/paying-for-college2/understanding-your-financial-aid-offer/offer/?iped=133465&pid=5287&oid=ABCDEABCDEABCDEABCDEABCDEABCDEABCDEABCDE&totl=45000&tuit=38976&hous=3000&book=650&tran=500&othr=500&pelg=1500&schg=2000&stag=2000&othg=100&ta=3000&mta=3000&gib=3000&wkst=3000&parl=14000&perl=3000&subl=15000&unsl=2000&ppl=1000&gpl=1000&prvl=3000&prvi=4.55&prvf=1.01&insl=3000&insi=4.55&inst=8&leng=30)
6. When you're done, don't forget to clean up your `main` branch if you did the `git checkout`. - switching to `main` and doing a `git clean -i` will get it done!

## Checklist
